pkg/pillar/cmd/upgradeconverter/upgradeconverter.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
```

Comment:
With the dis-aggregation effort going on, this script no longer belongs in pillar and should be moved to https://github.com/lf-edge/eve/tree/master/pkg/storage-init instead.

Reply:
Hi Roman, Can you explain why this should be in storage init? UpgradeConvertier is tightly tied to the pillar components - including using their data structures. I don;t see how this belongs outside of pillar..

Reply:
Because @kalyan-nidumolu pillar no longer "owns" persist the way it used to. Pillar is now simply a "customer" of a state in /persist but how that state is arranged is managed by storage-init (and soon vault -- that hasn't happened yet).

Inversely, whatever the "upgrader" code does -- well, it now has customers outside of pillar container (lisp is today's example, but soon there will be more). You can't "sequence" things in device-steps.sh anymore.

Makes sense?

Reply:
@rvs To move it we have to move the decrypt of /persist/config to storage-init as well.